diffedit.md

```diff
@@ -0,0 +1,752 @@
+    uncond_embeddings = text_encoder(uncond_input.input_ids.to("cuda"))[0].half()
+
+    return torch.cat([uncond_embeddings, *text_embeddings])
+```
+
+To get an idea of what the tokenised text looks like:
+
+
+```python
+tokenizer(["horse"], padding="max_length", max_length=tokenizer.model_max_length, truncation=True, return_tensors="pt")
+```
+
+
+
+
+    {'input_ids': tensor([[49406,  4558, 49407, 49407, 49407, 49407, 49407, 49407, 49407, 49407,
+             49407, 49407, 49407, 49407, 49407, 49407, 49407, 49407, 49407, 49407,
+             49407, 49407, 49407, 49407, 49407, 49407, 49407, 49407, 49407, 49407,
+             49407, 49407, 49407, 49407, 49407, 49407, 49407, 49407, 49407, 49407,
+             49407, 49407, 49407, 49407, 49407, 49407, 49407, 49407, 49407, 49407,
+             49407, 49407, 49407, 49407, 49407, 49407, 49407, 49407, 49407, 49407,
+             49407, 49407, 49407, 49407, 49407, 49407, 49407, 49407, 49407, 49407,
+             49407, 49407, 49407, 49407, 49407, 49407, 49407]]), 'attention_mask': tensor([[1, 1, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
+             0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
+             0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
+             0, 0, 0, 0, 0]])}
+
+
+
+We can see above that our prompt of `"horse"` has been converted into different tokens:
+
+- `49406` is equal to start
+- `4558` is equal to `"horse"`
+- `49407` is equal to end/padding
+
+The tensor is padded to be a particular length that the model expects, but an attention mask is also passed in which tells the model which tokens it should pay attention to, where `1` means it should pay attention to it and `0` means ignore. In this case it is the first three tokens only.
+
+
+```python
+prompts = [original_prompt, new_prompt]
+combined_embeddings = create_text_embeddings(prompts)
+```
+
+## Noising the original image
+
+To add noise to our original image to the correct level, we will utilise the scheduler. The scheduler typically manages noise throughout the image creation process for diffusion models, and it can also be used to manually add a specific level of noise to an image.
+
+First we need to set up the scheduler, and provide it with the configuration values that our diffusion model expects. We have used a DDIM scheduler, which is the type of scheduler that is recommended by the DiffEdit paper as it allows for more consistent results between the original image and the edited image.
+
+We'll define this setup as a function as we'll be using it again later when we generate the DiffEdit image. The argument `num_inference_steps` is the total number of steps that our model would take to fully denoise an image. With the version of Stable Diffusion and the scheduler we have chosen, 50 steps usually works well.
+
+
+```python
```

Comment:
A general comment is that while the notebook format encourages lots of code cells, typically in blog posts there's more prose than code. I would be tempted to trim some of the code blocks (maybe 50%).

Reply:
I still think there's a few too many. There's quite a few places where you initialise/assign variables. In most cases the variable name is sufficient to follow the explanation in the text. The specific values would only be required if you were trying to reproduce the post and the link to the repo should cover off that use case.